### PR TITLE
moved subscriber creation to subscription creation moment from login moment

### PR DESCRIPTION
**Pull request by laki88**

### CODE REVIEW

File: components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIStoreHostObject.java [hunk 1]
```diff
@@ -1599,6 +1600,30 @@ public static boolean jsFunction_addAPISubscription(Context cx,
             trustedIdp = args[7].toString();
         }
 
+        APIConsumer apiConsumer = getAPIConsumer(thisObj);
```
Comment (dinusha-dilrukshi) [inline components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIStoreHostObject.java:1598]: Keep line 1603-1625 in a separate private method (#addSubscriber) and call it here.

File: components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIStoreHostObject.java [hunk 2]
```diff
@@ -1599,6 +1600,30 @@ public static boolean jsFunction_addAPISubscription(Context cx,
...
+            try {
+                int tenantId =
+                        ServiceReferenceHolder.getInstance().getRealmService().getTenantManager()
+                                              .getTenantId(
+                                                      MultitenantUtils.getTenantDomain(userId));
+                subscriber.setTenantId(tenantId);
+                apiConsumer.addSubscriber(subscriber);
+            } catch (AppManagementException e) {
+                handleException("Error while adding the subscriber" + subscriber.getName(), e);
+                return false;
```
Comment (dinusha-dilrukshi) [inline components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIStoreHostObject.java]: handleException method throws a new exception. Hence having return statement after that make no sense.

File: components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIStoreHostObject.java [hunk 3]
```diff
@@ -1599,6 +1600,30 @@ public static boolean jsFunction_addAPISubscription(Context cx,
...
+            } catch (Exception e) {
+                handleException("Error while adding the subscriber" + subscriber.getName(), e);
+                return false;
```
Comment (dinusha-dilrukshi) [inline components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIStoreHostObject.java]: Same here.

File: components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIStoreHostObject.java (hunk 1 above)
Comment (laki88) [inline components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIStoreHostObject.java:1598]: resolved with 5234bc12ffc86f55a6e418db5818b6879a8bdea7

**Pull request merged**